sdk/elixir/dagger_codegen/lib/dagger/codegen/elixir_generator/renderer.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Dagger.Codegen.ElixirGenerator.Renderer do
+  @moduledoc """
+  Provides functions to render small part of Elixir code.
+  """
+
+  alias Dagger.Codegen.ElixirGenerator.Formatter
+  alias Dagger.Codegen.Introspection.Types.Type
+  alias Dagger.Codegen.Introspection.Types.Field
+
+  @doc """
+  Render the string.
+
+  Uses multiline string when newline is detected.
+  """
+  def render_string(s) do
+    if String.contains?(s, "\n") do
+      [
+        "\"\"\"",
+        ?\n,
+        s,
+        ?\n,
+        "\"\"\""
+      ]
+    else
+      [?", s, ?"]
+    end
+  end
+
+  @doc "Render atom value."
+  def render_atom(string), do: ":#{string}"
+
+  @doc """
+  Render moduledoc from `type`.
+
+  Uses the module name when no description in the type.
+  """
+  def render_moduledoc(type)
+
+  def render_moduledoc(%Type{description: desc, name: name} = type)
+      when desc == "" or is_nil(desc) do
+    # Prevent ExDoc ignore documentation when moduledoc were not found.
+    render_moduledoc(%{type | description: Formatter.format_module(name)})
+  end
+
+  def render_moduledoc(%Type{} = type) do
+    [
+      "@moduledoc",
+      ~c" ",
+      render_string(type.description)
+    ]
+    |> IO.iodata_to_binary()
+  end
+
+  @doc """
+  Render function document.
+  """
+  def render_doc(field_or_enum)
+
+  def render_doc(%{description: ""}), do: ""
+
+  def render_doc(%{description: description}) do
+    ["@doc", ~c" ", render_string(Formatter.format_doc(description))]
+    |> IO.iodata_to_binary()
+  end
+
+  @doc """
+  Render function deprecation message.
+  """
+  def render_deprecated(field)
+
+  def render_deprecated(%Field{deprecation_reason: ""}), do: ""
+  def render_deprecated(%Field{deprecation_reason: nil}), do: ""
+
+  def render_deprecated(%Field{deprecation_reason: reason}) do
```

Comment:
We have a convention that deprecation reason messages can reference a field (most likely the field that should be used instead), wrapped in backticks (e.g.,  ``use `newField` instead``). The reason for this is that SDKs can extract whatever's inside the backticks from the deprecation reason and re-format it according to its own naming conventions. 

Example in Python:

https://github.com/dagger/dagger/blob/795a73595d8dd0c14bc9b71e6a137852e7e0e9ac/sdk/python/src/dagger/_codegen/generator.py#L664-L673

https://github.com/dagger/dagger/blob/795a73595d8dd0c14bc9b71e6a137852e7e0e9ac/docs/docs-graphql/schema.graphqls#L2142

https://github.com/dagger/dagger/blob/795a73595d8dd0c14bc9b71e6a137852e7e0e9ac/sdk/python/src/dagger/client/gen.py#L5696

Reply:
Currently, I've done field conversion by transform content in backquote into snakecase. But the hard part is to link to referenced function in Elixir, we need the arity which's the number of argument of that function. So it's now become a little bit tricky that we need to traverse introspection to find the exact number. 😢 

 I think it would be great to follow up this issue on the next PR.

Reply:
Reference to issue https://github.com/dagger/dagger/issues/6939

Reply:
Oh, now I see that's replaced in all descriptions, not just deprecations. We've avoided referencing other fields in the description in the core API because of the wrong naming convention in other languages, but I think that's no longer the case. It's easy to creap in. 

We've limited this to deprecation reasons because it's more common in field descriptions to want something in backquotes other than being a field name. Doesn't seem like there's a case like this today though, but that can change. It's smart to only match valid identifiers, but I can't rule out a name in backquotes that is a valid identifier, but meant for something else other than an API field.

Changing the casing is enough. When I said "reference", I wasn't referring to anything special in code, even though Python does generate a prefix that allows it to be hyperlinked correctly in the reference docs. But the convention here is just to rename, having IDE autocomplete in mind.